client/me/application-password-item/index.jsx

```diff
@@ -12,44 +12,28 @@ import { localize } from 'i18n-calypso';
  * Internal dependencies
  */
 import Button from 'components/button';
+import { deleteApplicationPassword } from 'state/application-passwords/actions';
 import { errorNotice } from 'state/notices/actions';
 import { recordGoogleEvent } from 'state/analytics/actions';
 
 class ApplicationPasswordsItem extends React.Component {
-	state = {
-		removingPassword: false,
-	};
-
 	handleRemovePasswordButtonClick = () => {
+		const { password } = this.props;
+
 		this.props.recordGoogleEvent( 'Me', 'Clicked on Remove Application Password Button' );
-		this.removeApplicationPassword();
+		this.props.deleteApplicationPassword( parseInt( password.ID, 10 ) );
```

Comment:
Why `parseInt`? `ID` should pretty safely be an integer thanks the the schemas ([#](https://github.com/Automattic/wp-calypso/blob/02c9c213446ab47682fb7732df93458b3439c43a/client/state/data-layer/wpcom/me/two-step/application-passwords/schema.js#L9), [#](https://github.com/Automattic/wp-calypso/blob/02c9c213446ab47682fb7732df93458b3439c43a/client/state/application-passwords/schema.js#L6)).

Looks like this is maintained from the previous version. Can we rip it out?

Reply:
Good call - it's indeed no longer necessary; removed in https://github.com/Automattic/wp-calypso/pull/23926/commits/e00f74c1c96709e424998fb4f1e63b40c436907d